### PR TITLE
refactor tarian

### PR DESCRIPTION
**Pull request by pratikjagrut**
This pull request brings significant changes by revamping the CLIs using cobra. It also incorporates a new logging and error mechanism, enhancing the overall functionality and user experience.

### CODE REVIEW

**Comment by pratikjagrut** [comment]
`charts / lint-test` workflow is failing because it's using an image from ghcr registry and changes in this PR makes it obsolete so ignore it for now.

**Comment by devopstoday11** [comment]
@andylibrian , please check this PR. 

@pratikjagrut  @pratikjagrut,  have you tested all the changes before submitting this PR? I'm just confirming.

File: .github/workflows/anchore.yml [hunk 1]
```diff
@@ -1,28 +1,19 @@
...
-# and parameters, see https://github.com/anchore/scan-action. For more
-# information on Anchore's container image scanning tool Grype, see
-# https://github.com/anchore/grype
-name: Anchore Container Scan
+name: Vulnerability Scan
 
 on:
-  push:
-    branches: [ "main" ]
+  release:
```
Comment (andylibrian) [inline .github/workflows/anchore.yml:4]: @pratikjagrut What's the reason behind this change? I'm only concerned if we do this on release, we don't run it often. We only know if it fails / succeeds only on releasing, which could be a surprising failure.

Comment (andylibrian) [reply .github/workflows/anchore.yml:4]: Oh I realize there's a PR too, so never mind

File: .goreleaser.yml [hunk 2]
```diff
@@ -120,18 +147,18 @@ dockers:
...
-    skip_push: false
-    build_flag_templates:
-    - "--pull"
-    - "--label=org.opencontainers.image.created={{.Date}}"
-    - "--label=org.opencontainers.image.title={{.ProjectName}}"
-    - "--label=org.opencontainers.image.revision={{.FullCommit}}"
-    - "--label=org.opencontainers.image.version={{.Version}}"
-    - "--label=org.opencontainers.image.source={{ .GitURL }}"
+  # - ids:
+  #   - tarianctl
```
Comment (andylibrian) [inline .goreleaser.yml:151]: @pratikjagrut is this disabled intentionally? Any context?

File: dev/config/monitored-pod/configmap.yaml [hunk 3]
```diff
@@ -4,7 +4,7 @@ data:
     <!DOCTYPE html>
     <html>
     <head>
-    <title>Welcome to nginx!</title>
+    <title>Welcome-updated-updated-updated-updated to nginx!</title>
```
Comment (andylibrian) [inline dev/config/monitored-pod/configmap.yaml:7]: @pratikjagrut I believe this is committed by accident. Let's revert this. This file is updated during the k8s-test, but it shouldn't be committed.

File: dev/config/monitored-pod/configmap.yaml [hunk 4]
```diff
@@ -14,7 +14,7 @@ data:
     </style>
     </head>
     <body>
-    <h1>Welcome to nginx!</h1>
+    <h1>Welcome-updated-updated-updated-updated to nginx!</h1>
```
Comment (andylibrian) [inline dev/config/monitored-pod/configmap.yaml:17]: Same with above. This file is updated during the k8s-test, but it shouldn't be committed.

File: .goreleaser.yml (hunk 2 above)
Comment (pratikjagrut) [reply .goreleaser.yml:151]: Yes. Tarianctl is a CLI tool so I think we don't need an image for that. We can only release the binary of it and that should suffice.

File: dev/config/monitored-pod/configmap.yaml (hunk 3 above)
Comment (pratikjagrut) [reply dev/config/monitored-pod/configmap.yaml:7]: Oh yes.

File: dev/config/monitored-pod/configmap.yaml (hunk 4 above)
Comment (pratikjagrut) [reply dev/config/monitored-pod/configmap.yaml:17]: Yes.

File: .goreleaser.yml (hunk 2 above)
Comment (andylibrian) [reply .goreleaser.yml:151]: I added this not too long ago. My intention was to improve the execution time of chart test. My plan was for chart test, we can just use latest container images from the repo, without building local images. Feel free to let me know your feedback.

However, I haven't finished that. Currently in `.github/workflows/charts.yaml`, we need to tarianctl to run the test but we build the whole components. My plan was to docker pull tarianctl and copy the executable from the image to the current directory.

**Review by andylibrian — APPROVED** [review]

**Pull request merged**